lib/system/freertos/xlnx_common/zynqmp_aarch64/sys.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023, Xilinx Inc. and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	freertps/xlnx_common/zynqmp_aarch64/sys.h
+ * @brief	freertos zynqmp_aarch64 system primitives for libmetal.
+ */
+
+#ifndef __METAL_FREERTOS_SYS__H__
+#error "Include metal/sys.h instead of metal/freertos/@PROJECT_MACHINE@/sys.h"
+#endif
+
+#include <metal/system/@PROJECT_SYSTEM@/xlnx_common/sys.h>
+#include "xscugic.h"
+
+#ifndef __METAL_FREERTOS_ZYNQMP_XLNX_COMMON_AARCH64_SYS__H__
+#define __METAL_FREERTOS_ZYNQMP_XLNX_COMMON_AARCH64_SYS__H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef METAL_INTERNAL
+
+#define XLNX_MAXIRQS XSCUGIC_MAX_NUM_INTR_INPUTS
+
+static inline void sys_irq_enable(unsigned int vector)
+{
+	XScuGic_EnableIntr(XPAR_SCUGIC_0_DIST_BASEADDR, vector);
```

Comment:
Since this is under freertos directory is it possible to use FreeRTOS APIs to do this instead of using XILINX APIS. For example to enable irq, freertos must be providing some kind of API.

Reply:
I looked in Xilinx BSP port of FReeRTOS and using vPortEnableInterrupt() might work. I will have to test.